### PR TITLE
Fix: Transformer encoder - apply LayerNorm after residual

### PR DESCRIPTION
**Pull request by sonali-kumari1**
This PR fixes the Transformer encoder to properly apply Layer Normalization after the residual connections, in accordance with the specification from the [Attention is All You Need" (Vaswani et al., 2017)](https://arxiv.org/abs/1706.03762) paper. 

Previously, Layer Normalization was incorrectly applied before the residual connection, which violated the design outlined in Section 3.1 of "Attention is All You Need," where it states: "the output of each sub-layer is`LayerNorm(x + Sublayer(x))`". 

This update ensures that Layer Normalization is applied after the residual addition in both the attention and feed-forward sub-layers. 

Fixes : [#1256 ](https://github.com/keras-team/keras-io/issues/1256)

### CODE REVIEW

**Comment by sachinprasadhs** [comment]
There is no issue with the existing implementation.

The implementation in https://keras.io/examples/timeseries/timeseries_classification_transformer is also correct, it is an improvement to the original paper by having Layer Normalization inside the block so that the training will be stable.

Here is a blog explaining the difference between both the approaches https://sh-tsang.medium.com/review-pre-ln-transformer-on-layer-normalization-in-the-transformer-architecture-b6c91a89e9ab

May be you can just add a note explaining this variation.

**Comment by sachinprasadhs** [comment]
Thanks, could you please make the changes to .md and .ipynb files as well, you don't have to render all the files, you can just make manual changes to those files.

**Comment by sachinprasadhs** [comment]
Could you please run black format to the changed files

**Review by sachinprasadhs — APPROVED** [review]
